bin/porta

```diff
@@ -489,15 +508,23 @@ module Porta
 
     # this would hopefully work with podman 4 and docker but not tested
     def container_opts_local_hosts(*hosts)
-      return "" unless need_to_add_internal_hosts?
+      if macos? && !podman?
+        # should work for --add-host but not tested for --dns
+        gw_spec = 'host-gateway'
+      else
+        gw_spec = container_internal_ip
+      end
 
-      hosts << "host.containers.internal" << "host.docker.internal"
-      gw_ip = container_internal_host || 'host-gateway'
-      host_access = slirp4netns_arg.dup
-      if File.read("/etc/resolv.conf") =~ /nameserver\s+127\.0\.0\.1$/
-        host_access += " --dns=#{gw_ip}"
+      unless need_to_add_internal_hosts?
+        hosts << "host.containers.internal" << "host.docker.internal"
+        host_access = slirp4netns_arg.dup
+        if File.read("/etc/resolv.conf") =~ /nameserver\s+127\.0\.0\.1$/
+          host_access += " --dns=#{gw_spec}" # this is not needed and may break old podman in `host` network mode
+        end
       end
-      hosts.map { |h| "--add-host=#{h}:#{gw_ip} " }.join + host_access
+
+      hosts << "master-account.3scale.localhost"
```

Comment:
I haven't tried this yet, but I'll try to answer some questions:

1. APIcast has 2 ways of fetching configs, it can either fetch for a single provider, if configured with a provider URL, or for all providers (tenants), if configured with a Master URL. The paths in these two ways are different, I think. So, it is enough when apicast can talk to master, it should be able to fetch the configs properly.
2. I actually use the standard local domains :) But there is also this issue on the repo: https://github.com/3scale-labs/porta-dev-tools/issues/1 So, it's an RFE :)

Reply:
After reading https://github.com/3scale-labs/porta-dev-tools/issues/1, I think the best approach is to get rid `porxy` and add the master domain to our [settings](https://github.com/3scale-labs/porta-dev-tools/blob/45df70346b387b5ed21457249b00b99541020539/config/examples/settings.yml), using `master-account.3scale.localhost` as default.

Reply:
RFE - request for enhancement

I think using that domain as well poses the same problem - it has to be resolvable to a useful IP by apicast.